R/mstl.R

```diff
@@ -449,8 +468,15 @@ stlm <- function(y, s.window=13, robust=FALSE, method=c("ets", "arima"), modelfu
   seascols <- grep("Seasonal", colnames(stld))
   allseas <- rowSums(stld[, seascols, drop = FALSE])
 
-  fits <- fitted(fit) + allseas
-  res <- residuals(fit)
+  if (is.na(fit$x[1])){
```

Comment:
Why condition on the first observation being missing?

Reply:
I was using `method = "arima"`, which in my call to `stlm`. This results in `auto.arima` being used to fit the time series. When that is done, the `fit` element of the model object is a vector whose first elements corresponds to the first non-missing value in the `x` element, or the input data, which I think can be traced to [this line](https://github.com/robjhyndman/forecast/blob/fbd25c66d079a43f4e59023f23edc54707387ec7/R/newarima2.R#L131). This leads to a difference in length of the `fitted(fit)` and `allseas`, and I added this conditional statement as a simple fix to avoid the resulting error.